### ISSUE TITLE
use smaller resolution

### ISSUE THREAD

**Issue by lukasHoel** [baaivision/NOVA#11]
Hi thanks for your help, I got another question:

can we change the image (video) resolution to be a bit lower?
I tried to make the 1024x1024 T2I model generate a lower resolution like this:

```
import torch
from diffnext.pipelines import NOVAPipeline
from diffnext.models.embeddings import PosEmbed

model_id = "BAAI/nova-d48w768-sdxl1024"
model_args = {"torch_dtype": torch.float16, "trust_remote_code": True}
pipe = NOVAPipeline.from_pretrained(model_id, **model_args)
pipe = pipe.to("cuda")

def update_resolution(self, height: int, width: int):
    """Update the image resolution."""

    # update values in transformer config
    s = self.transformer.config.image_stride
    pi = self.transformer.image_encoder.patch_embed.patch_size
    pv = self.transformer.video_encoder.patch_embed.patch_size
    self.transformer._internal_dict["image_size"] = self.transformer.config.image_size = (height, width)
    self.transformer.image_encoder.image_size = (height // s, width // s)
    self.transformer.video_encoder.image_size = (height // s, width // s)
    self.transformer._internal_dict["image_base_size"] = self.transformer.config.image_base_size = (height // (s * pi), width // (s * pi))
    self.transformer._internal_dict["video_base_size"] = self.transformer.config.video_base_size = (self.transformer.config.video_base_size[0], height // (s * pv), width // (s * pv))

    # update values in pos_emb
    def update_pos_emb(pos_emb, h, w):
        if pos_emb is None or isinstance(pos_emb, torch.nn.Identity):
            return
        elif isinstance(pos_emb, PosEmbed):
            pos_emb.space_embed = None
            pos_emb.base_h = h
            pos_emb.base_w = w
        else:
            raise NotImplementedError(f"Unsupported positional embedding: {pos_emb}")

    # update video pos embed
    update_pos_emb(self.transformer.video_pos_embed, *self.transformer.config.video_base_size[1:])
    update_pos_emb(self.transformer.video_encoder.pos_embed, *self.transformer.config.video_base_size[1:])

    # update image pos embed
    update_pos_emb(self.transformer.image_pos_embed, *self.transformer.config.image_base_size)
    update_pos_emb(self.transformer.image_encoder.pos_embed, *self.transformer.config.image_base_size)

update_resolution(pipe, 256, 256)

prompt = "a shiba inu wearing a beret and black turtleneck."
image = pipe(prompt).images[0]
    
image.save("shiba_inu.jpg")
```

But the generated image does not look good. Since the pos embedding is not trainable I thought maybe it is compatible to different resolutions. Am I right that it would require further finetuning to work or is there another way?

**Comment by PhyscalX** [comment]
Our released models use absolute position embeddings, which involves attention collapse when #tokens > 1K. (i.e., 512x512)
In our internal developments, NOVA models with RoPE could reduce height or width by up to 50%.

**Comment by lukasHoel** [comment]
Thanks for the response! Do you have advice/code how to change to RoPE the most elegant way with this codebase?

**Comment by PhyscalX** [comment]
Set  ``rotary_pos_embed`` to ``true``,  in ``/path/to/nova-d48w768-sdxl1024/transformer/config.json``.

**Comment by lukasHoel** [comment]
I will now try to finetune the model with RoPE (do you plan to release your models with RoPE as well?).
Can you tell me in your internal experiments, did you finetune the whole model (full fine-tuning) or add LoRAs and only finetune those? I wonder if finetuning on a smaller-scale dataset + with only LoRAs would be sufficient to understand the RoPE embeddings.

**Comment by PhyscalX** [comment]
We fine-tuned the whole model with a high learning rate, thus LoRAs may not work.
